executable-monitor/action.yml

```diff
@@ -19,14 +20,59 @@ inputs:
     description: 'Number of times to re-launch the binary to check for success.'
     required: false
     default: 0
+  success-exit-code:
+    description: 'Exit status that indicates that the executable completed successfully. Required if --success-line is not used.'
+    required: false
+    default: ""
 
 runs:
   using: "composite"
   steps: 
-      - name: Install dependencies
-        run: pip install -r $GITHUB_ACTION_PATH/requirements.txt
+      - name: Install Dependencies
+        run: |
+          # Install Dependencies
+          echo "::group::Install Dependencies"
+          pip install -r $GITHUB_ACTION_PATH/requirements.txt
+          echo "::endgroup::"
+          if [ "$?" = "0" ]; then
+            echo -e "\033[32;3mInstalled all dependencies\033[0m"
+            exit 0
+          else
+            echo -e "\033[32;31mDependencies Install failed...\033[0m"
+            exit 1
+          fi
         shell: bash
-      - name: Run executable with monitoring script
+      - name: Run Executable with Monitoring
         run: |
-          python3 $GITHUB_ACTION_PATH/executable-monitor.py --exe-path=${{ inputs.exe-path }} --timeout-seconds=${{ inputs.timeout-seconds }} --success-line="${{ inputs.success-line }}" --log-dir=${{ inputs.log-dir }} --retry-attempts=${{ inputs.retry-attempts }}
+          # Run Executable with Monitoring
+          echo "::group::Executable Output"
+          echo "inputs.success-exit-code:${{ inputs.success-exit-code }}\n"
+          echo "inputs.success-line:${{ inputs.success-line }}\n"
+          optArgs=" " 
+          if [ "${{ inputs.success-exit-code }}" = "" ] && [ "${{ inputs.success-line }}" = "" ]; then
+            echo "::endgroup::"
+            echo -e "\033[32;31mDid not supply an input of success-line or success-exit-code to search for\033[0m"
+            exit 1
+          fi
+          if [ "${{ inputs.success-exit-code }}" != "" ]; then
+             optArgs="--success-exit-code=${{ inputs.success-exit-code }}$optArgs "
+          fi
+          
+          if [ "${{ inputs.log-dir }}" != "" ]; then
+            optArgs="--log-dir=${{ inputs.log-dir}} $optArgs"
+          fi
+          echo "optArgs= $optArgs"
+          if [ "${{ inputs.success-line }}" != "" ]; then
+            python3 $GITHUB_ACTION_PATH/executable-monitor.py --exe-path=${{ inputs.exe-path }} --timeout-seconds=${{ inputs.timeout-seconds }} $optArgs --success-line="${{ inputs.success-line}}"
+          else
+            python3 $GITHUB_ACTION_PATH/executable-monitor.py --exe-path=${{ inputs.exe-path }} --timeout-seconds=${{ inputs.timeout-seconds }} $optArgs
+          fi
+          echo "::endgroup::"
+          if [ "$?" = "0" ]; then
```

Comment:
Shouldn't we check against `${{ inputs.success-exit-code }}`?
Or am I not understanding this correctly?

Reply:
The executable monitor itself does this check, where from here I want to know if the python application exited correctly

Reply:
Aha, I see.